### ISSUE TITLE
Convert lat/lon to x/y for polar stereographic projections

### ISSUE THREAD

**Issue by USGSsakins** [USGS-Astrogeology/CartoCosmos#5]
When displaying north or south poles, coordinates need to be converted from x/y to lat/lon and back.

**Comment by USGSsakins** [comment]
@thareUSGS This was something we had to do for the polar WMS layers in the past since their extents were defined as meter by meter x/y coordinates, is this still the case?

**Comment by thareUSGS** [comment]
Yes. This is for any projection besides the simple "degree" geocentric/geographic case. We must convert from meters to degrees, which requires the projection equation (its parameters) and the radius (unless we have access to the proj library which can do this for us).

**Comment by thareUSGS** [comment]
This example is a little confusing but does the map projection in code.
https://openlayers.org/en/latest/examples/wms-custom-proj.html?q=proj

I think we can actually define a Mars (other) projection in "proj" (using a proj string) and have the code do the conversion for us. perhaps something like

```
...
import {addProjection, addCoordinateTransforms, transform} from 'ol/proj';
import {register} from 'ol/proj/proj4';
import proj4 from 'proj4';

// Mars 2000 as a sphere, just in degrees, ocentric
// https://spatialreference.org/ref/iau2000/49900/
proj4.defs('IAU2000:49900', '+proj=longlat ' +
    '+a=3396190 +b=3396190 ' +
    '+no_defs');

// Mars 2000, North Pole (+lat_0=90), for South Pole use (+lat_0=-90)
// https://spatialreference.org/ref/iau2000/49918/
proj4.defs('IAU2000:49918', '+proj=stere ' +
    '+lat_0=90 +lon_0=0 ' +  
    '+k=1 +x_0=0 +y_0=0 ' +
    '+a=3396190 +b=3396190 ' +
    '+units=m +no_defs');
register(proj4);

// for Mars polar, from: https://astrowebmaps.wr.usgs.gov/webmapatlas/Layers/maps.html
var imageExtent = [-2357032, -2357032, 2357032, 2357032];

// test some meters to latlon -- maybe, not tested
  var myLat = 85
  var myLon = 120
  var myXY = transform([myLon,myLat], 'IAU2000:49900', 'IAU2000:49918')
  var myX = myXY[0]
  var myY = myXY[1]

  // very approximate calculation of projection extent
  var extent = applyTransform(
    [bbox[1], bbox[2], bbox[3], bbox[0]], fromLonLat);

// make a map -- maybe
// need Mars North Pole WMS layers from:
// https://astrowebmaps.wr.usgs.gov/webmapatlas/Layers/maps.html
var map = new Map({
  layers: [
...
  ],
  target: 'map',
  view: new View({
    projection: 'IAU2000:49918',
    center: [0, 0],
    zoom: 2,
    extent: imageExtent
  })
});
```

**Comment by jlaura** [comment]
It probably makes sense to use: http://proj4js.org since we have access to it?

**Comment by USGSsakins** [comment]
It does make sense to use proj4.js, however keep in mind that OpenLayers and Leaflet may already be including proj4.js.  There may be problems with importing conflicting Javascript implementations into the same namespace.  Can you access the version in the mapping library?

**Comment by USGSsakins** [comment]
@thareUSGS We can get the IAU approved radii values for the `proj4.defs` call with the info in the web map atlas.  I would "fake" the other fields (ie. definition name 'IAU2000:49918") until we convert to a catalog server and include the missing fields as part of the "target/layer" definitions to dynamically create the definitions.  That certainly saves us having to modify code when a new body is added.

**Comment by thareUSGS** [comment]
It will be tempting to use a transformation function, 
https://github.com/USGS-Astrogeology/AstroWebMaps/blob/c5aa66b92211a98711a4cbca5d052bd132fab7ae/js/astrowebmaps/Helpers/AstroGeometry.js#L641

but I am hoping we can get ProjJS to do this for us (see above).

**Comment by thareUSGS** [comment]
> can get the IAU approved radii values for the

They will be available in proj6, we are skipping proj4 since it doesn't handle ocentric/ographic as well as the updated version,

**Comment by kaitlyndlee** [comment]
We will need to import proj4 separately to define proj-strings. Then to get OpenLayers to use the projections, we use the register function.
